app/views/samples_reports/show.haml

```diff
@@ -331,5 +330,62 @@
     }
   }
 
+  function getSvg(chartId, chartName){
+    var svgData = document.getElementById(chartId);
+    applyInline(svgData);
+    svgData = svgData.outerHTML;
+    return svgData;
+  }
+
+  async function printPDF() {
+    var context = getUrlParameter('context');
+    
+    var measured_signal_svg = encodeURIComponent(getSvg("barchart", "measured_signal"));
+    var specific_svg;
+    if ("#{@purpose}" == "Challenge")
+      specific_svg = encodeURIComponent(getSvg("rocchart", "roc_curve"));
+    else
+      specific_svg = encodeURIComponent(getSvg("linechart", "limit_of_detection"));
+
+    // if purpose is challenge, then we need the live threshold calculations
+    if (#{@purpose.to_json} == "Challenge") {
+      var threshold = document.getElementById("threshold").value;
+      var url = '/samples_reports/update_threshold?context='+context+"&samples_report_id=#{@samples_report.id}&threshold="+threshold;
+      await fetch(url)
+          .then((response) => response.json())
+          .then((ret) => {
+            var params = {
+              id: #{@samples_report.id},
+              threshold: threshold,
+              auc: #{auc(roc_curve(@samples_report))},
+              threshold_tpr: (ret.confusion_matrix.true_positive || 0) / ((ret.confusion_matrix.true_positive || 0) + (ret.confusion_matrix.false_negative || 0)),
+              threshold_fpr: (ret.confusion_matrix.false_positive || 0) / ((ret.confusion_matrix.false_positive || 0) + (ret.confusion_matrix.true_negative || 0)),
+              measured_signal_svg: measured_signal_svg,
+              specific_svg: specific_svg
+            };
+            var print_url = '/samples_reports/print'; 
+            postObjectToUri(print_url, params, context);
+        })
+    }
+    else {
+      var params = {
+        id: #{@samples_report.id},
+        measured_signal_svg: measured_signal_svg,
+        specific_svg: specific_svg
+      };
+      var print_url = '/samples_reports/print'; 
+      postObjectToUri(print_url, params, context);
+    }
+  }
```

Comment:
Couldn't we get all these params right from the actual SamplesReport on the backend? Then we'd only have to send the SVG charts to the backend, and that's it.

If it's the result of the confusion matrix calculation, then it could be re-calculated on the backend from the selected threshold in the `print` action, which avoids the above `update_threshold` request.